### DISCUSSION TITLE
Error on open /admin

### DISCUSSION THREAD

**Discussion by csmaniottojr** [fief-dev/fief#219] [Q&A] [answered]
I got the following error when accessing the "/admin/auth/login" endpoint locally. I'm testing Fief with docker-compose locally, before installing in an Amazon EC2 instance.

### Trace

```
fief-fief-server-1  | Exception in ASGI application
fief-fief-server-1  | Traceback (most recent call last):
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
fief-fief-server-1  |     result = await app(  # type: ignore[func-returns-value]
fief-fief-server-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
fief-fief-server-1  |     return await self.app(scope, receive, send)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 276, in __call__
fief-fief-server-1  |     await super().__call__(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 336, in _sentry_patched_asgi_app
fief-fief-server-1  |     return await middleware(scope, receive, send)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 140, in _run_asgi3
fief-fief-server-1  |     return await self._run_app(scope, lambda: self.app(scope, receive, send))
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 189, in _run_app
fief-fief-server-1  |     raise exc from None
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 184, in _run_app
fief-fief-server-1  |     return await callback()
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
fief-fief-server-1  |     await self.middleware_stack(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
fief-fief-server-1  |     raise exc
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
fief-fief-server-1  |     await self.app(scope, receive, _send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief/middlewares/x_forwarded_host.py", line 26, in __call__
fief-fief-server-1  |     return await self.app(scope, receive, send)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 140, in _run_asgi3
fief-fief-server-1  |     return await self._run_app(scope, lambda: self.app(scope, receive, send))
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 150, in _run_app
fief-fief-server-1  |     raise exc from None
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 147, in _run_app
fief-fief-server-1  |     return await callback()
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 230, in _sentry_exceptionmiddleware_call
fief-fief-server-1  |     await old_call(self, scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
fief-fief-server-1  |     raise exc
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
fief-fief-server-1  |     await self.app(scope, receive, sender)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
fief-fief-server-1  |     raise e
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
fief-fief-server-1  |     await route.handle(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 443, in handle
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 276, in __call__
fief-fief-server-1  |     await super().__call__(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 336, in _sentry_patched_asgi_app
fief-fief-server-1  |     return await middleware(scope, receive, send)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 140, in _run_asgi3
fief-fief-server-1  |     return await self._run_app(scope, lambda: self.app(scope, receive, send))
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 150, in _run_app
fief-fief-server-1  |     raise exc from None
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 147, in _run_app
fief-fief-server-1  |     return await callback()
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
fief-fief-server-1  |     await self.middleware_stack(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
fief-fief-server-1  |     raise exc
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
fief-fief-server-1  |     await self.app(scope, receive, _send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/sessions.py", line 86, in __call__
fief-fief-server-1  |     await self.app(scope, receive, send_wrapper)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/gzip.py", line 24, in __call__
fief-fief-server-1  |     await responder(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/gzip.py", line 44, in __call__
fief-fief-server-1  |     await self.app(scope, receive, self.send_with_gzip)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief/middlewares/csrf.py", line 23, in __call__
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief/middlewares/security_headers.py", line 17, in __call__
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 230, in _sentry_exceptionmiddleware_call
fief-fief-server-1  |     await old_call(self, scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
fief-fief-server-1  |     raise exc
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
fief-fief-server-1  |     await self.app(scope, receive, sender)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/starlette.py", line 131, in _create_span_call
fief-fief-server-1  |     return await old_call(app, scope, new_receive, new_send, **kwargs)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
fief-fief-server-1  |     raise e
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 718, in __call__
fief-fief-server-1  |     await route.handle(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 276, in handle
fief-fief-server-1  |     await self.app(scope, receive, send)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 66, in app
fief-fief-server-1  |     response = await func(request)
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 237, in app
fief-fief-server-1  |     raw_response = await run_endpoint_function(
fief-fief-server-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
fief-fief-server-1  |     return await dependant.call(**values)
fief-fief-server-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief/apps/dashboard/routers/auth.py", line 24, in login
fief-fief-server-1  |     url = await fief.auth_url(
fief-fief-server-1  |           ^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief_client/client.py", line 764, in auth_url
fief-fief-server-1  |     openid_configuration = await self._get_openid_configuration()
fief-fief-server-1  |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/fief_client/client.py", line 1003, in _get_openid_configuration
fief-fief-server-1  |     response = await client.send(request)
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_client.py", line 1620, in send
fief-fief-server-1  |     response = await self._send_handling_auth(
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_client.py", line 1648, in _send_handling_auth
fief-fief-server-1  |     response = await self._send_handling_redirects(
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_client.py", line 1685, in _send_handling_redirects
fief-fief-server-1  |     response = await self._send_single_request(request)
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_client.py", line 1722, in _send_single_request
fief-fief-server-1  |     response = await transport.handle_async_request(request)
fief-fief-server-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_transports/default.py", line 352, in handle_async_request
fief-fief-server-1  |     with map_httpcore_exceptions():
fief-fief-server-1  |   File "/usr/local/lib/python3.11/contextlib.py", line 155, in __exit__
fief-fief-server-1  |     self.gen.throw(typ, value, traceback)
fief-fief-server-1  |   File "/usr/local/lib/python3.11/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
fief-fief-server-1  |     raise mapped_exc(message) from exc
fief-fief-server-1  | httpx.ConnectError: All connection attempts failed
```

### docker-compose.yml:

```yml
version: "3"

services:
  fief-server:
    image: ghcr.io/fief-dev/fief:latest
    command: fief run-server --port 80
    ports:
      - 8000:80
    env_file:
      - .env

  fief-worker:
    image: ghcr.io/fief-dev/fief:latest
    command: fief run-worker -p 1 -t 1
    env_file:
      - .env

  postgres:
    image: postgres:alpine
    ports:
      - "5433:5432"
    environment:
      - POSTGRES_PASSWORD=fief
      - POSTGRES_USER=fief
      - POSTGRES_DB=fief
    volumes:
      - postgres-data:/var/lib/postgresql/data

  redis:
    image: redis:alpine
    command: redis-server --save 60 1 --loglevel warning
    volumes:
      - redis-data:/data

volumes:
  redis-data:
  postgres-data:
```

### .env

```
SECRET=XXX
FIEF_CLIENT_ID=XXX
FIEF_CLIENT_SECRET=XXX
ENCRYPTION_KEY=XXX=
PORT=8000
ROOT_DOMAIN=localhost:8000
FIEF_DOMAIN=localhost:8000
FIEF_MAIN_USER_EMAIL=cesar.smaniotto@spotlar.com.br
FIEF_MAIN_USER_PASSWORD=z^k1P&KICOS27k*x2C

DATABASE_TYPE=POSTGRESQL
DATABASE_HOST=postgres
DATABASE_PORT=5432
DATABASE_USERNAME=fief
DATABASE_PASSWORD=fief
DATABASE_NAME=fief

REDIS_URL=redis://redis:6379
```

**Comment by fief-bailiff** [comment]
Hail, @csmaniottojr 👋

I've noticed you shared secret values: `SECRET`, `FIEF_CLIENT_ID`, `FIEF_CLIENT_SECRET`, `ENCRYPTION_KEY`. Those are highly sensitive and you should keep them secret.

For your security, I've taken the liberty to replace them with dummy values.

**Comment by fief-bailiff** [comment]
Hail, @csmaniottojr 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
There's a problem with the port definition: the environment variable `PORT` and the port on which you open the server must match. So, if the server command is `fief run-server --port 80`, then you should have `PORT=80` in `.env`.

It's a bit hard to follow, because the port is different inside and outside the container. I would suggest to be consistent everywhere, if that's possible for you (e.g. 8000 everywhere or 80 everywhere).

**Reply by csmaniottojr** [reply]
Nice, I thought that adjusting the port mapping in docker-compose would be enough. I changed the code to keep ports consistent and worked. Thanks, @frankie567
